flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/config/BasicAsyncSinkWriterConfiguration.java

```diff
@@ -0,0 +1,162 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.base.sink.writer.config;
+
+import org.apache.flink.connector.base.sink.writer.strategy.AIMDScalingStrategy;
+import org.apache.flink.connector.base.sink.writer.strategy.CongestionControlRateLimitingStrategy;
+import org.apache.flink.connector.base.sink.writer.strategy.RateLimitingStrategy;
+
+/**
+ * BasicAsyncSinkWriterConfiguration with defaults that work well for streaming sources/sinks like
+ * Kinesis Data Streams and Kinesis Firehose.
+ */
+public class BasicAsyncSinkWriterConfiguration implements AsyncSinkWriterConfiguration {
+    private final int maxBatchSize;
+    private final int maxInFlightRequests;
+    private final int maxBufferedRequests;
+    private final long maxBatchSizeInBytes;
+    private final long maxTimeInBufferMS;
+    private final long maxRecordSizeInBytes;
+    private final RateLimitingStrategy rateLimitingStrategy;
+
+    private BasicAsyncSinkWriterConfiguration(
+            int maxBatchSize,
+            int maxInFlightRequests,
+            int maxBufferedRequests,
+            long maxBatchSizeInBytes,
+            long maxTimeInBufferMS,
+            long maxRecordSizeInBytes,
+            RateLimitingStrategy rateLimitingStrategy) {
+        this.maxBatchSize = maxBatchSize;
+        this.maxInFlightRequests = maxInFlightRequests;
+        this.maxBufferedRequests = maxBufferedRequests;
+        this.maxBatchSizeInBytes = maxBatchSizeInBytes;
+        this.maxTimeInBufferMS = maxTimeInBufferMS;
+        this.maxRecordSizeInBytes = maxRecordSizeInBytes;
+        this.rateLimitingStrategy = rateLimitingStrategy;
+    }
+
+    public int getMaxBatchSize() {
+        return maxBatchSize;
+    }
+
+    public int getMaxInFlightRequests() {
+        return maxInFlightRequests;
+    }
+
+    public int getMaxBufferedRequests() {
+        return maxBufferedRequests;
+    }
+
+    public long getMaxBatchSizeInBytes() {
+        return maxBatchSizeInBytes;
+    }
+
+    public long getMaxTimeInBufferMS() {
+        return maxTimeInBufferMS;
+    }
+
+    public long getMaxRecordSizeInBytes() {
+        return maxRecordSizeInBytes;
+    }
+
+    public RateLimitingStrategy getRateLimitingStrategy() {
+        return rateLimitingStrategy;
+    }
+
+    public static BasicAsyncSinkWriterConfigurationBuilder builder() {
+        return new BasicAsyncSinkWriterConfigurationBuilder();
+    }
+
+    /** Builder for {@link BasicAsyncSinkWriterConfiguration}. */
+    public static class BasicAsyncSinkWriterConfigurationBuilder {
+
+        private int maxBatchSize = 500;
+        private int maxInFlightRequests = 50;
+        private int maxBufferedRequests = 10_000;
+        private long maxBatchSizeInBytes = 5 * 1024 * 1024;
+        private long maxTimeInBufferMS = 5_000;
+        private long maxRecordSizeInBytes = 1024 * 1024;
```

Comment:
I do not think it is a good idea to have defaults in the base implementation. This also makes the validation less useful, users could develop sinks without considering these values and result in non-performant sink.

Reply:
Removed defaults, and made it mandatory in builder by implementing interface